src/Controls/src/Core/Handlers/Items2/iOS/CarouselViewController2.cs

```diff
@@ -132,12 +133,34 @@ private protected override async void AttachingToWindow()
 		{
 			base.AttachingToWindow();
 			Setup(ItemsView);
+			// Refresh the current visible item to catch any ItemsSource changes that occurred on other pages
+			// This ensures that updates made on other pages are reflected when navigating back
+			if (_wasDetachedFromWindow)
+			{
+				RefreshVisibleItems();
+			}
+			_wasDetachedFromWindow = false;
 			// if we navigate back on NavigationController LayoutSubviews might not fire.
 			await UpdateInitialPosition();
 		}
 
+		void RefreshVisibleItems()
+		{
+			if (CollectionView is null || ItemsSource?.ItemCount == 0)
+			{
+				return;
+			}
+			//Get current visible item index paths to ensure proper refresh of carousel items
+			var indexPaths = CollectionView.IndexPathsForVisibleItems;
+			if (indexPaths?.Length > 0)
+			{
+				CollectionView.ReloadItems(indexPaths);
```

Comment:
Could under some condition the carousel briefly shows the old cell and then reloads it? (flickering)

Reply:
Based on testing, there are no flashing or flickering issues